[FLINK-31248] Improve documentation for append-only table (#562)
apache/flink-table-store: docs/content/docs/concepts/append-only-table.md

type: docs
aliases:
- /concepts/append-only-table.html
---
<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

# Append Only Table

By specifying `'write-mode' = 'append-only'` when creating the table, user creates an append-only table.

You can only insert a complete record into the table. No delete or update is supported and you cannot define primary keys.
This type of table is suitable for use cases that do not require updates (such as log data synchronization).

## Bucketing

> You also need to define bucket number for Append-only table, see [Bucket]({{< ref "docs/concepts/basic-concepts#bucket" >}}).

Reviewer: ```suggestion
You can also define bucket number for Append-only table, see [Bucket]({{< ref "docs/concepts/basic-concepts#bucket" >}}).
```

Users don't "need to" define bucket number. They can always use the default bucket number, which is 1.

It is recommended that you set the `bucket-key` field. Otherwise, the data will be hashed according to the whole row,
and the performance will be poor.

## Streaming Read Order

For streaming reads, records are produced in the following order:

* For any two records from two different partitions
  * If `scan.plan-sort-partition` is set to true, the record with a smaller partition value will be produced first.
  * Otherwise, the record with an earlier partition creation time will be produced first.
* For any two records from the same partition and the same bucket, the first written record will be produced first.
* For any two records from the same partition but two different buckets, different buckets are processed by different tasks, there is no order guarantee between them.

## Compaction

> By default, the sink node will automatically perform compaction to solve the small file problem. The following options

Reviewer: ```suggestion
By default, the sink node will automatically perform compaction to control the number of files. The following options
```
